### PR TITLE
Eradicate now() calls

### PR DESCRIPTION
**Pull request by fenek**
Proposed changes include:

* Replacement of all `now()` calls with proper equivalents from `p1_time_compat` (that will be eventually replaced with `erlang` module in the future).
* Replacement of `now()` calls in integration tests with `os:timestamp()`.

Some explanations:

* Whenever I was sure that `now()` was used to generate unique values or was hashed, I have replaced it with `unique_integer`, since it is recommended by OTP docs.
* `timestamp()` was my default choice where I wasn't sure if it's safe to use any other function.
* `mod_bosh_socket` now uses `monotonic_time/1`, because it used `now()` for time difference calculation, so I've chosen function recommended by OTP docs.
* I've used `os:timestamp()` in `mod_commands` because its output is used later together with `p1_time_compat:os_system_time(micro_seconds)`, so I considered it appropriate to stick to OS time.
* I've used `os:timestamp()` in integration tests in order not to introduce new dependency here. This function is good enough in tests, as they do not have as strict time "quality" requirements, as server code does.

### CODE REVIEW

File: apps/ejabberd/src/ejabberd_s2s_out.erl [hunk 1]
```diff
@@ -1111,8 +1111,7 @@ get_addr_port(Server) ->
         {ok, #hostent{h_addr_list = AddrList}} ->
             %% Probabilities are not exactly proportional to weights
             %% for simplicity (higher weigths are overvalued)
-            {A1, A2, A3} = now(),
-            random:seed(A1, A2, A3),
+            random:seed(randoms:good_seed()),
             case (catch lists:map(
                           fun({Priority, Weight, Port, Host}) ->
```
Comment (gadgetci) [inline apps/ejabberd/src/ejabberd_s2s_out.erl]: According to **Elvis**:
> The expression on line 1116 and column 27 is nested beyond the maximum level of 3.

File: apps/ejabberd/src/extauth.erl [hunk 2]
```diff
@@ -134,8 +134,7 @@ call_port(Server, Msg) ->
 
 -spec random_instance(pos_integer()) -> non_neg_integer().
 random_instance(MaxNum) ->
-    {A1, A2, A3} = now(),
-    random:seed(A1, A2, A3),
+    random:seed(randoms:good_seed()),
```
Comment (gadgetci) [inline apps/ejabberd/src/extauth.erl]: According to **Compiler**:
> Warning: random:seed/1: the 'random' module is deprecated; use the 'rand' module instead

File: apps/ejabberd/src/extauth.erl [hunk 3]
```diff
@@ -134,8 +134,7 @@ call_port(Server, Msg) ->
...
     random:uniform(MaxNum) - 1.
```
Comment (gadgetci) [inline apps/ejabberd/src/extauth.erl]: According to **Compiler**:
> Warning: random:uniform/1: the 'random' module is deprecated; use the 'rand' module instead

File: apps/ejabberd/src/ejabberd_s2s_out.erl [hunk 4]
```diff
@@ -1111,8 +1111,7 @@ get_addr_port(Server) ->
         {ok, #hostent{h_addr_list = AddrList}} ->
             %% Probabilities are not exactly proportional to weights
             %% for simplicity (higher weigths are overvalued)
-            {A1, A2, A3} = now(),
-            random:seed(A1, A2, A3),
+            random:seed(randoms:good_seed()),
```
Comment (gadgetci) [inline apps/ejabberd/src/ejabberd_s2s_out.erl]: According to **Compiler**:
> Warning: random:seed/1: the 'random' module is deprecated; use the 'rand' module instead

File: test.disabled/ejabberd_tests/tsung/mam_con.xml [hunk 5]
```diff
@@ -58,7 +58,7 @@
       <!-- Returns a random message id of 24 hours age -->
       <![CDATA[
         fun() ->
-            Now = now_to_microseconds(now()),
+            Now = now_to_microseconds(os:timestamp()),
```
Comment (michalwski) [inline test.disabled/ejabberd_tests/tsung/mam_con.xml]: 😆 I completely forgot we still have some tsung scenarios in tests. I wonder if we could remove them now. I don't remember when they were used, especially now when we have amoc and tide.

Comment (fenek) [inline test.disabled/ejabberd_tests/tsung/mam_con.xml]: They still can be useful for some user but I absolutely don't mind removing them. :)

File: apps/ejabberd/src/mod_mam_utils.erl [hunk 6]
```diff
@@ -178,7 +178,7 @@ microseconds_to_datetime(MicroSeconds) when is_integer(MicroSeconds) ->
 generate_message_id() ->
     {ok, NodeId} = ejabberd_node_id:node_id(),
     %% Use monotone function here.
-    encode_compact_uuid(now_to_microseconds(now()), NodeId).
+    encode_compact_uuid(now_to_microseconds(p1_time_compat:timestamp()), NodeId).
```
Comment (michalwski) [inline apps/ejabberd/src/mod_mam_utils.erl]: We do need unique timestamp here. The whole `ID` is later used as `primary` key in DB.

**Comment by michalwski** [comment]
Thanks!

**Pull request merged**